api/tests/mocha/controllers/export-data-2.js

```diff
@@ -0,0 +1,50 @@
+require('chai').should();
+
+const controller = require('../../../controllers/export-data-2');
+
+describe('Export Data 2.0', () => {
+  describe('Extract fields', () => {
+    it('flattens fields', () => {
+      controller._flatten({
+        foo: 'fooVal',
+        bars: {
+          bar1: 'bar1Val',
+          smang: {
+            smong: 'smongVal'
+          }
+        }
+      }).should.deep.equal({
+        foo: 'fooVal',
+        'bars.bar1': 'bar1Val',
+        'bars.smang.smong': 'smongVal'
+      });
+    });
+    it('Converts null / undefined into empty string', () => {
+      controller._flatten({
+        foo: '',
+        bar: null,
+        smang: undefined
+      }).should.deep.equal({
+        foo: '',
+        bar: '',
+        smang: ''
+      });
+    });
+    it('Handles arrays', () => {
+      // TODO: do we want it to work this way? Does it matter?
+      controller._flatten({
```

Comment:
I'm unsure about this. IDK if we want to do this, or instead convert an array into a `string`. Or if it matters.

Reply:
True. I wonder if we use arrays in any of our fields. I suppose it's at least a possibility. I'll look into how to deal with this.

FWIW we're not yet using a CSV export library, which would probably be helpful here, as it would deal with doing proper quoting and the like. This may be something I need to look into.

Reply:
Actually we don't need a library, we just need to quote everything and escape quotes.

Reply:
did this